libraries/cmake/source/librdkafka/CMakeLists.txt

```diff
@@ -138,11 +157,16 @@ function(generateLibrdkafkaxx)
 
   target_include_directories(thirdparty_librdkafka_cxx PRIVATE
     "${CMAKE_CURRENT_SOURCE_DIR}/config/dummy"
-    "${library_root}"
+	"${library_root}"
+  )
+
+  target_compile_definitions(thirdparty_librdkafka_cxx PUBLIC
+  LIBRDKAFKA_STATICLIB
   )
 
   target_include_directories(thirdparty_librdkafka_cxx SYSTEM INTERFACE
     "${CMAKE_CURRENT_BINARY_DIR}/include"
+	"${CMAKE_CURRENT_SOURCE_DIR}/../zstd/src/lib/common"
```

Comment:
Please don't depend in this way on the path/presence of another library.
The `thirdparty_zstd` target should publish that path.

I'm noticing now a couple of other issues though:
1) In the zstd library we publish the `src/lib` path, because it contains the zstd.h header, but that's not enough, we should also take **some** headers from the `src/lib/common` folder.
Official packages seems to include zbuff.h, zdict.h, zstd_errors.h.
2) Both the zstd library and librdkafka compile xxhash.c and I fear there's a collision between symbols, meaning that the linker will take one or the other implementation, which might not be the same version that the library expects.
This needs investigation. I see that xxhash permits you to pass a `XXH_PRIVATE_API` define which should in theory solve this, but looking at what it does, I don't get how it is supposed to (at least not in the case of compiling a static library).

Reply:
1.    zstd_errors.h , pain point indeed , that solution could work
2.    xxhash.c idk about that , it didn't produce any collisions, or warnings in my runtime, i`ll  try to investigate more (linker symbol collision , nice topic for discussion :) )

Reply:
> 2. xxhash.c idk about that , it didn't produce any collisions, or warnings in my runtime, i`ll  try to investigate more (linker symbol collision , nice topic for discussion :) )

Ok to be fair this is pre-existing issue which I noticed now and was thinking to resolve in one go, but we can deal with this in another PR.

Basically here you won't see any warning because the linker does its job "correctly", meaning that if it finds that a symbol is already imported, it won't import it again from another static library.
This means that that depending on the order the libraries are linked in, the selected implementation of xxhash changes.
Though this might not be a major issue, since I expect xxhash to be compatible between versions, nonetheless it would be better if each library uses the implementation it expects.

Reply:
thirdparty_zstd updated